core/services/src/main/java/org/exoplatform/services/attachments/service/AttachmentServiceImpl.java

```diff
@@ -282,6 +301,40 @@ public Attachment getAttachmentById(String attachmentId, SessionProvider session
     }
   }
 
+  @Override
+  public void moveAttachmentToNewPath(long userIdentityId,
+                                      String attachmentId,
+                                      String newPathDrive,
+                                      String newPath) throws Exception {
+    if (userIdentityId <= 0) {
+      throw new IllegalAccessException("User identity must be positive");
+    }
+    if (StringUtils.isEmpty(attachmentId)) {
+      throw new IllegalArgumentException("Attachment id is mandatory");
+    }
+    if (StringUtils.isEmpty(newPathDrive)) {
+      throw new IllegalArgumentException("New destination path's drive must not be empty");
+    }
+
+    SessionProvider sessionProvider = sessionProviderService.getSystemSessionProvider(null);
+    String workspace = repositoryService.getCurrentRepository().getConfiguration().getDefaultWorkspaceName();
+    Session session = sessionProvider.getSession(workspace, repositoryService.getCurrentRepository());
+    try {
+      Node attachmentNode = session.getNodeByUUID(attachmentId);
+      Node destNode = getParentFolderNode(workspace, session, newPathDrive, newPath);
+      String destPath = destNode.getPath().concat("/").concat(attachmentNode.getName());
+      session.move(attachmentNode.getPath(), destPath);
+      attachmentNode.refresh(true);
+      session.save();
+    } catch (Exception e) {
+      LOG.error("Error while trying to move attachment node with id {} to the new path {} " + attachmentId, newPath, e);
```

Comment:
please avoid swallowing exception and throw the exception to REST endpoint that will handle it to return the adequate HTTP status code to end user. (No need to log here neither)